cromwellApiClient/src/main/scala/cromwell/api/CromwellClient.scala

```diff
@@ -30,13 +30,30 @@ class CromwellClient(val cromwellUrl: URL, val apiVersion: String, val credentia
   lazy val submitEndpoint = s"$cromwellUrl/api/workflows/$apiVersion"
   // Everything else is a suffix off the submit endpoint:
   lazy val batchSubmitEndpoint = s"$submitEndpoint/batch"
-  private def workflowSpecificEndpoint(workflowId: WorkflowId, endpoint: String) = s"$submitEndpoint/$workflowId/$endpoint"
-  def abortEndpoint(workflowId: WorkflowId): String = workflowSpecificEndpoint(workflowId, "abort")
-  def statusEndpoint(workflowId: WorkflowId): String = workflowSpecificEndpoint(workflowId, "status")
-  def metadataEndpoint(workflowId: WorkflowId): String = workflowSpecificEndpoint(workflowId, "metadata")
-  def outputsEndpoint(workflowId: WorkflowId): String = workflowSpecificEndpoint(workflowId, "outputs")
-  def labelsEndpoint(workflowId: WorkflowId): String = workflowSpecificEndpoint(workflowId, "labels")
-  def logsEndpoint(workflowId: WorkflowId): String = workflowSpecificEndpoint(workflowId, "logs")
+
+  /**
+    * @param args an optional map of HTTP arguments which will be added to the URL
+    */
+  private def workflowSpecificGetEndpoint(workflowId: WorkflowId, endpoint: String, args: Option[Map[String, List[String]]] = None) = {
+    // Converts the args map to an HTTP argument string, e.g. "foo=bar&baz=qux"
+    def argMapToString(argMap: Map[String, List[String]]): Iterable[String] = {
+      for {
+        arg <- argMap
+        value <- arg._2
+      } yield s"${arg._1}=$value"
+    }
+
+    val argString = args.map(argMapToString)
+    val url = s"$submitEndpoint/$workflowId/$endpoint"
+    argString.fold(url)(a => s"$url?$a")
```

Comment:
ToL: I think it would be better to use a java `URL` or `URI` for those instead of raw strings that we build manually

Reply:
Oh, wrong `CromwellClient`. I have no idea why these aren't URLs, but I certainly didn't have any vague memories of trying as I didn't do this :) I'll take a look.

I'll reiterate my hatred for this class :)

Reply:
I was just coming back to say that I might ignore this one :)